include/graft_utility.hpp

```diff
@@ -36,38 +85,74 @@ namespace graft
         TSList(TSList&& other) = delete;
         TSList& operator=(TSList const& other) = delete;
 
-        void pushFront(T const& value)
+        void pushFront(T const& value, ch::seconds ttl = ch::seconds(0))
         {
-            std::unique_ptr<node> new_node(new node(value));
+            std::unique_ptr<node> new_node(new node(value, ttl));
 
             std::lock_guard<std::mutex> lk(head.m);
             new_node->next = std::move(head.next);
             head.next = std::move(new_node);
         }
 
-        void forEach(func f)
+        void insert(T const& value, ch::seconds ttl = ch::seconds(0))
+        {
+            std::unique_ptr<node> new_node(new node(value, ttl));
+            std::unique_lock<std::mutex> lk(head.m);
+
+            fixup_tail(&head);
+
+            node* current = &head;
+            while (node* next = current->next.get())
+            {
+                std::unique_lock<std::mutex> next_lk(next->m);
+                lk.unlock();
+
+                if (*new_node <= *current->next)
+                {
```

Comment:
I think it is incorrect to use current->next, because you unlocked current and another thread can modify current->next